samples: matter: Enable Partition Manager and DFU for nrf54l15. (#14895)
nrfconnect/sdk-nrf: samples/matter/template/README.rst

    :start-after: matter_door_lock_sample_build_with_dfu_start
    :end-before: matter_door_lock_sample_build_with_dfu_end

.. matter_template_nrf54l15_build_with_dfu_start

The Device Firmware Upgrade (DFU) for the nRF54L15 PDK is exclusively available for the ``release`` build configuration and is limited to the use of internal MRAM for storage.

> This implies that both the currently running firmware and the new firmware to be updated are stored within the device's internal flash memory.

Reviewer: ```suggestion
This means that both the currently running firmware and the new firmware to be updated must be stored within the device's internal flash memory.
```

> As of now, there is no support for utilizing external flash memory for this purpose.

Reviewer: ```suggestion
Currently, there is no support for utilizing external flash memory for this purpose.
```
"Currently" is used more in our documentation, though "as of now" isn't wrong.

> To build the sample with DFU support, you should use the ``-DFILE_SUFFIX=release_internal`` flag in your CMake build command.

Reviewer: ```suggestion
To build the sample with DFU support, use the ``-DFILE_SUFFIX=release_internal`` flag in your CMake build command.
```

> An example command to build the sample with support for OTA DFU only:

Reviewer: ```suggestion
The following is an example command to build the sample with support for OTA DFU only:
```

.. code-block:: console

>     west build -b nrf54l15pdk_nrf54l15_cpuapp -- -DFILE_SUFFIX=release_internal

Reviewer 2: Do we want to use `FILE_SUFFIX`? I mean, other configurations were not ported yet, so maybe let's stick with the `CONF_FILE` solution to be consistent? Supporting two different ways of configuration for different boards will be slightly confusing.

Author: OK, we can use CONF_FILE, you're right, we should be consistent even if FILE_SUFIX works here though.

> If you want to build the sample with support for both OTA DFU and Serial and SMP DFU, use the following command

Reviewer: ```suggestion
If you want to build the sample with support for both OTA DFU and Serial and SMP DFU, use the following command:
```

> If you want to build the sample with support for both OTA DFU and Serial and SMP DFU, use the following command:

Reviewer 2: Why does it mention Serial?

    :start-after: matter_door_lock_sample_build_with_dfu_start
    :end-before: matter_door_lock_sample_build_with_dfu_end

.. matter_template_nrf54l15_build_with_dfu_start

The Device Firmware Upgrade (DFU) for the nRF54L15 PDK is exclusively available for the ``release`` build configuration and is limited to the use of internal MRAM for storage.
This means that both the currently running firmware and the new firmware to be updated must be stored within the device's internal flash memory.
Currently, there is no support for utilizing external flash memory for this purpose.

To build the sample with DFU support, use the ``-DCONF_FILE=prj_release.conf`` flag in your CMake build command.

The following is an example command to build the sample with support for OTA DFU only:

.. code-block:: console

    west build -b nrf54l15pdk_nrf54l15_cpuapp -- -DCONF_FILE=prj_release.conf

If you want to build the sample with support for both OTA DFU and SMP DFU, use the following command:

.. code-block:: console

    west build -b nrf54l15pdk_nrf54l15_cpuapp -- -DCONF_FILE=prj_release.conf -DCONFIG_CHIP_DFU_OVER_BT_SMP=y

> You can disable DFU support for the ``release`` build configuration and in this way double application memory space by setting the :kconfig:option:`CONFIG_CHIP_DFU_OVER_BT_SMP`, and :kconfig:option:`CONFIG_CHIP_OTA_REQUESTOR` Kconfig options to ``n``.

Reviewer: ```suggestion
You can disable DFU support for the ``release`` build configuration to double available application memory space.
Do this by setting the :kconfig:option:`CONFIG_CHIP_DFU_OVER_BT_SMP`, and :kconfig:option:`CONFIG_CHIP_OTA_REQUESTOR` Kconfig options to ``n``.
```
